### PR TITLE
Added matching of maintainer to user via github username

### PR DESCRIPTION
**Pull request by jackhughesweb**
Closes exercism/exercism#3835

### CODE REVIEW

File: app/models/git/github_profile.rb [hunk 1]
```diff
@@ -26,4 +26,9 @@ def initialize(user)
     @avatar_url = user.avatar_url
     @link_url = user.html_url
   end
+
+  def user_present?
+    user && user.id
+  end
+
```
Comment (iHiD) [inline app/models/git/github_profile.rb]: Mind removing this extra line?

**Comment by iHiD** [comment]
@jackhughesweb Once we've merged this how do we run it for existing mentors?

**Comment by jackhughesweb** [comment]
@iHiD The changes are inside `upsert_maintainer`, which will insert or update maintainers

**Comment by iHiD** [comment]
@jackhughesweb Merging this but I don't think it will update for old maintainers unless they change something. @kntsoriano Could you work out how to run this on the servers pls?

**Pull request merged**